src/org/zalando/stups/kio/api.clj

```diff
@@ -144,8 +144,13 @@
                   :description         nil
                   :specification_type  nil
                   :publicly_accessible false
-                  :criticality_level   2}]
-    (require-write-authorization request (:team_id application))
+                  :criticality_level   2}
+        existing_application (load-application application_id db)]
+
+    (if (nil? existing_application)
```

Comment:
can be simplified to `(if existing-application ...`).
Of course "then" and "else" form need to be switched.